2023-risks/m10-insufficient-binary-protection.md

```diff
@@ -0,0 +1,102 @@
+
+In general, fully compiled apps like iOS apps are less susceptible to reverse engineering and code tampering than higher level bytecode as in Android apps. 
+
+Especially popular apps are likely to be manipulated and redistributed through app stores. Detecting and removing these apps is offered by specialized companies but also possible with certain detection and reporting mechanisms within the apps.
+
+Note that there are no fully reliable mechanisms to prevent binary attacks. Defending against them is an arms race between the developers investing in countermeasures and attackers who break these measures. So the question to be answered for each app is: How much effort should be put into measures against binary attacks?
+
+
+# Technical Impacts	
+
+**Impact Moderate**
+
+If secrets leak, they must be replaced quickly throughout the system, which is hard if the secrets are hardcoded in the app. Information leakage from the binary also has potential to reveal security vulnerabilities in the backend. Yet, manipulation has more impact on the technical soundness of a system. 
+
+By manipulation of the binaries, attackers could change how apps work arbitratily, for example to their own benefit or to disturb the backends, if they are insufficiently hardened against malicious requests.
+
+
+# Business Impacts
+	
+**Application / Business Specific** 
+
+Leakage of API-keys for commercial APIs or similar can cause significant costs if they are misused on a large scale. Even more, if intellectual property, like algorithms or KI models that have been developed with great effort, become public, the business model of the app developers may be threatened.
+
+The same holds for apps that are tampered to remove license checks or to publish their functionality with a competing app.
+
+Great reputational damage could arise in particular for popular apps that get redistributed with malicious code.
+
+
+# Am I Vulnerable To 'Insufficient Binary Protection'?
+
+All apps are vulnerable to binary attacks. Binary attacks can become particularly harmful if the app has sensitive data or algorithms hardcoded in its binary or if it is very popular. If there are additional protective measures, like obfuscation, encoding of secrets in native code (for Android) or similar, successful attacks become harder to achieve but never impossible. 
+
+Whether the app is sufficiently secure after all depends on the business impact that different binary attacks could have. The more motivating it is for attackers and the greater the impact would be, the more effort should be put into protection. Hence, "vulnerability" to binary attacks is highly specific to the given app.
+
+For a quick check, developers could inspect their own app binaries using similar tools as attackers would use. There are many free or affordable tools, like IDA Pro, Hopper, otool, apktool and Ghidra that are also quite easy to use and well documented.
+
+
```

Comment:
I would avoid mentioning commercial tools here. I think would not consider IDA Pro as affordable.